argilla/src/argilla/client.py

```diff
@@ -260,11 +295,30 @@ def __init__(self, client: "Argilla") -> None:
         self._client = client
         self._api = client.api.datasets
 
-    def __call__(self, name: str, workspace: Optional[Union["Workspace", str]] = None) -> Optional["Dataset"]:
+    @overload
+    def __call__(self, name: str, workspace: Optional[Union["Workspace", str]] = None) -> Optional["Dataset"]: ...
+
+    @overload
+    def __call__(self, id: Union[UUID, str]) -> "Dataset": ...
+
+    def __call__(
+        self, name: str = None, workspace: Optional[Union["Workspace", str]] = None, id: Union[UUID, str] = None
+    ) -> Optional["Dataset"]:
+        if not (name or id):
+            raise ArgillaError("One of 'name' or 'id' must be provided")
+
+        if name and id:
+            warnings.warn("Only one of 'name' or 'id' must be provided. Using 'id'")
+            name = None
+
+        if id is not None:
+            if not isinstance(id, UUID):
+                id = UUID(id)
```

Comment:
Can't we move some of this code to a parent class to avoid having duplicate code?